packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2473,6 +2492,14 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
     }
 
     private updateDocumentDirtyState(dirty: boolean) {
+        if (this.attachState !== AttachState.Attached) {
```

Comment:
This seems unrelated

Reply:
Yes. You can help me get rid of it from this PR :) - it exists as stand-alone fixes - https://github.com/microsoft/FluidFramework/pull/11823.
It's required for this change to succeed (indirectly, as some tests were failing due to bugs, and these asserts help discover these bugs and prevent them in the future).